connexion/operations/swagger2.py

```diff
@@ -224,30 +234,95 @@ def _nested_example(self, schema):
         except KeyError:
             raise
 
-    @property
-    def body_schema(self):
+    def body_schema(self, content_type: str = None) -> dict:
         """
         The body schema definition for this operation.
         """
-        return self.with_definitions(self.body_definition).get("schema", {})
+        body_definition = self.body_definition(content_type)
+        return self.with_definitions(body_definition).get("schema", {})
 
-    @property
-    def body_definition(self):
+    def body_definition(self, content_type: str = None) -> dict:
         """
         The body complete definition for this operation.
 
         **There can be one "body" parameter at most.**
 
         :rtype: dict
         """
-        body_parameters = [p for p in self.parameters if p["in"] == "body"]
-        if len(body_parameters) > 1:
-            raise InvalidSpecification(
-                "{method} {path} There can be one 'body' parameter at most".format(
-                    method=self.method, path=self.path
+        if content_type in FORM_CONTENT_TYPES:
+            form_parameters = [p for p in self.parameters if p["in"] == "formData"]
+            body_definition = self._transform_form(form_parameters)
+        else:
+            body_parameters = [p for p in self.parameters if p["in"] == "body"]
+            if len(body_parameters) > 1:
+                raise InvalidSpecification(
+                    "{method} {path} There can be one 'body' parameter at most".format(
+                        method=self.method, path=self.path
+                    )
                 )
-            )
-        return body_parameters[0] if body_parameters else {}
+            body_definition = body_parameters[0] if body_parameters else {}
+        return body_definition
+
+    def _transform_form(self, form_parameters: t.List[dict]) -> dict:
```

Comment:
Might be handy to add a unit test for this? Or do you think the other tests are sufficient?

Reply:
Yes a test would be good, I'll add one.